Implement redirect bridge to support COOP (#8118)
AzureAD/microsoft-authentication-library-for-js: lib/msal-browser/docs/login-user.md

## RedirectUri Considerations

**All authentication flows now require a dedicated redirect page** that implements the MSAL redirect bridge. This is necessary to support COOP (Cross-Origin-Opener-Policy) headers and enable secure communication between popup/iframe windows and the main application.

### Setting up the redirect page

Your `redirectUri` must point to a dedicated page that loads the redirect bridge script. This page should:

1. **Load the redirect bridge script** - This script handles communication with the main window

> 2. **Not implement any MSAL logic** - The redirect page should only run the bridge script

Reviewer: ```suggestion
2. **Not include any JavaScript except for bridge script** - The redirect page should only run the bridge script
```

    <script>
        msalRedirectBridge.sendRedirectPayloadToMainFrame();
    </script>
</body>
</html>
```

### Configuration

You can set the `redirectUri` globally in your MSAL configuration or on a per-request basis:

**Global configuration:**

```javascript
const msalConfig = {
    auth: {
        clientId: "your-client-id",
        authority: "https://login.microsoftonline.com/common",
        redirectUri: "http://localhost:3000/redirect"
    }
};

const msalInstance = new PublicClientApplication(msalConfig);
```

**Per-request configuration:**

```javascript
msalInstance.loginPopup({
    scopes: ["user.read"],
    redirectUri: "http://localhost:3000/redirect"
});
```

For more information and complete sample implementations, see:
- [React Router Sample](../../../samples/msal-react-samples/react-router-sample)
- [Express Sample](../../../samples/msal-browser-samples/ExpressSample)

## Handling popup `interaction_in_progress` errors

For popup flows, you can use the `overrideInteractionInProgress` flag to cancel a pending interaction and start a new one. This is useful for recovery scenarios where the user cancelled a popup or an interaction failed.

> > **Note:** This feature is **only available for popup flows** and is **not supported for redirect flows**. With the COOP (Cross-Origin-Opener-Policy) header, the traditional `window.opener` connection is severed, allowing popup windows to communicate with the main frame only via BroadcastChannel.

Reviewer: ```suggestion
> [!NOTE] 
> This feature is **only available for popup flows** and is **not supported for redirect flows**. With the COOP (Cross-Origin-Opener-Policy) header, the traditional `window.opener` connection is severed, allowing popup windows to communicate with the main frame only via BroadcastChannel.
```

> **WARNING**: Use with extreme caution! Setting this to `true` will forcefully cancel any pending popup interaction.

Reviewer: ```suggestion
> [!CAUTION] 
> Setting this to `true` will forcefully cancel any pending popup authentication request but **will not** close any open popups.
```

**When set to `true`:**

> - If another popup interaction is currently in progress, it will be forcefully cancelled

Reviewer: ```suggestion
- If another popup interaction is currently in progress, it will be forcefully cancelled but any open popups will not be closed
```

- The pending interaction will reject with an `interaction_in_progress_cancelled` error
- The new popup flow will proceed immediately

**Valid use cases:**
- Recovering from errors where the user cancelled a popup (popup was closed without completing auth)
- Implementing custom error recovery flows
- Providing a "retry" mechanism after a failed popup interaction

**Default:** `false`

### Important: Only Use on Button Click

**Do NOT automatically retry** when catching an `interaction_in_progress` error. The override should **only** be triggered by an explicit user action (such as clicking a "Retry" button). Automatically overriding interactions can lead to:
- Race conditions between multiple authentication flows
- Unexpected cancellations of legitimate authentication attempts

> - Poor user experience with authentication flows starting and stopping unexpectedly

Reviewer: ```suggestion
- Poor user experience with authentication flows starting and stopping unexpectedly
- Many open popups that will not resolve to successful authentication responses
```

> [!NOTE] 
> This feature is **only available for popup flows** and is **not supported for redirect flows**. With the COOP (Cross-Origin-Opener-Policy) header, the traditional `window.opener` connection is severed, allowing popup windows to communicate with the main frame only via BroadcastChannel.

> [!CAUTION] 
> Setting this to `true` will forcefully cancel any pending popup authentication request but **will not** close any open popups.

**When set to `true`:**
- If another popup interaction is currently in progress, it will be forcefully cancelled but any open popups will not be closed
- The pending interaction will reject with an `interaction_in_progress_cancelled` error
- The new popup flow will proceed immediately

**Valid use cases:**
- Recovering from errors where the user cancelled a popup (popup was closed without completing auth)
- Implementing custom error recovery flows
- Providing a "retry" mechanism after a failed popup interaction

**Default:** `false`

### Important: Only Use on Button Click

> **Do NOT automatically retry** when catching an `interaction_in_progress` error. The override should **only** be triggered by an explicit user action (such as clicking a "Retry" button). Automatically overriding interactions can lead to:

Reviewer 2: nit: May be we can use font colors and `RED` for warnings like this.
